components/activities/activities-tab/activities-tab.tsx

```diff
@@ -118,20 +93,54 @@ export function ActivitiesTabManager<
         <div className="list-header">
           ACTIVITY HISTORY
         </div>
-        {pastActivities?.length
-          ? (
-            <div className="list-group">
-              {pastActivities?.map((activity) => {
-                const ActivityTile = activity.sourceType
-                  ? activityListItems[activity.sourceType]
-                  : null;
-                return ActivityTile &&
-                  React.createElement(ActivityTile, { activity: activity.activity });
-              })}
-            </div>
-          )
-          : <div className="empty-list-group">No past activities</div>}
+        {/* activity li's will only rerender if their data changes */}
+        <Memo>
+          {() => (
+            <ActivityGroup
+              activityConnection$={activityAlertConnection$}
+              activityListItems={activityListItems}
+              isActive={false}
+              emptyStateMessage="No past activities"
+            />
+          )}
+        </Memo>
       </div>
     </div>
   );
 }
+
+function ActivityGroup<
+  ActivityListItemTypes,
+  SourceType extends StringKeys<ActivityListItemTypes>,
+  ActivityType extends ActivityListItemTypes[SourceType],
+>(
+  { activityConnection$, activityListItems, isActive, emptyStateMessage }: {
+    activityConnection$: Observable<ActivityConnection<ActivityType, SourceType>>;
+    activityListItems: ListItemMap<ActivityListItemTypes>;
+    isActive: boolean;
+    emptyStateMessage: string;
+  },
+) {
+  const activities = useSelector(() =>
+    activityConnection$?.nodes.get()?.filter((activity) =>
+      activity?.activity && isActive ? isActiveActivity(activity) : !isActiveActivity(activity)
+    )
+  );
+
+  return activities?.length
+    ? (
+      <div className="list-group">
+        {activities?.map((activity) => {
+          const ActivityListItem = activity.sourceType
+            ? activityListItems[activity.sourceType]
+            : null;
+
+          return ActivityListItem &&
+            React.createElement(ActivityListItem, {
```

Comment:
we should have a short standardish comment for why we use React.createElement whenever we do